src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -1212,7 +1213,7 @@ class RsFnInferenceContext(
             traitToCallee.keys.filterInScope(methodCall).forEach {
                 filtered += traitToCallee.getValue(it)
             }
-            filtered
+            if (TypeInferenceMarks.methodPickTraitsOutOfScope.hitOnFalse(filtered.isNotEmpty())) filtered else list
```

Comment:
Let's use simple if/else. This pretty simple condition starts to look very complex with this mark ^)

Reply:
Done